keras_cv/metrics/iou_3d.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 The KerasCV Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""IoU3D metric using a custom TF op."""
+
+import tensorflow as tf
+from tensorflow.python.framework import load_library
+from tensorflow.python.platform import resource_loader
+
+
+class IoU3D(tf.keras.metrics.MeanMetricWrapper):
+    """Implements the IoU metric for 3D upright rotated bounding boxes.
+
+    Note that this is implemented using a custom TensorFlow op. Initializing an
+    IoU3D object will attempt to load the binary for that op.
+
+    Boxes should have the format [center_x, center_y, center_z, dimension_x,
```

Comment:
Can we document a little bit the rational of this format choice in our python API?

I don't have stats on the popularity in the datasets, but If it is not the most popular native format we need to extend it our current 2d bbox converters to 3d.

E.g. pytorch3d ```# Assume inputs: boxes1 (M, 8, 3) and boxes2 (N, 8, 3)```

Please, check also the unified format in one of the largest 3d bbox dataset:
https://github.com/facebookresearch/omni3d/blob/main/DATA.md#data-usage

Reply:
I think it's likely that we'll want to add converters in the future. I've opened #947 for this and we can address it once we have a minimal 3D bbox pipeline working.